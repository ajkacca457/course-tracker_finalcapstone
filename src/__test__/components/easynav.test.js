import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Easynav from '../../components/easynav';

describe('Navbar Component', () => {
  it('renders correctly', () => {
    const navtree = TestRenderer
      .create(
      /* eslint-disable no-unused-vars */
        <BrowserRouter>
          {' '}
          <Easynav />
          {' '}
        </BrowserRouter>,
      )
      .toJSON();
    /* eslint-enable no-unused-vars */
    expect(navtree).toMatchSnapshot();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(<BrowserRouter><Easynav /></BrowserRouter>);
    const link = getByText('Home');
    expect(link).toBeInTheDocument();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(<BrowserRouter><Easynav /></BrowserRouter>);
    const link = getByText('Track');
    expect(link).toBeInTheDocument();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(<BrowserRouter><Easynav /></BrowserRouter>);
    const link = getByText('Add Course');
    expect(link).toBeInTheDocument();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(<BrowserRouter><Easynav /></BrowserRouter>);
    const link = getByText('Report');
    expect(link).toBeInTheDocument();
  });
});
