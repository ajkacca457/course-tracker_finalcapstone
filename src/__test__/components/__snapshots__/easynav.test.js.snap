// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders correctly 1`] = `
Array [
  " ",
  <div
    className="easynav"
  >
    <div
      className="navcont text-white bg-dark"
    >
      <div
        className="nav-item-easy"
      >
        <a
          className="text-white"
          href="/home"
          onClick={[Function]}
        >
          <i
            className="fas fa-home"
          />
          <p>
            Home
          </p>
        </a>
      </div>
      <div
        className="nav-item-easy"
      >
        <a
          className="text-white"
          href="/addcourse"
          onClick={[Function]}
        >
          <i
            className="fas fa-plus-circle"
          />
          <p>
            Add Course
          </p>
        </a>
      </div>
      <div
        className="nav-item-easy"
      >
        <a
          className="text-white"
          href="/courses"
          onClick={[Function]}
        >
          <i
            className="fas fa-chart-line"
          />
          <p>
            Track
          </p>
        </a>
      </div>
      <div
        className="nav-item-easy"
      >
        <a
          className="text-white"
          href="/progress"
          onClick={[Function]}
        >
          <i
            className="fas fa-chart-pie"
          />
          <p>
            Report
          </p>
        </a>
      </div>
    </div>
  </div>,
  " ",
]
`;
